a8c-files.php

```diff
@@ -58,6 +58,9 @@ function __construct() {
 
 		// ensure we always upload with year month folder layouts
 		add_filter( 'pre_option_uploads_use_yearmonth_folders', function( $arg ) { return '1'; } );
+
+		// ensure the correct upload URL is used even after switch_to_blog is called
+		add_filter( 'pre_option_upload_url_path', array( &$this, 'upload_url_path' ), 10, 1 );
```

Comment:
It's probably best that the mu-plugin always has the final say, but is there any risk or downside to using `option_upload_url_path` instead? The `pre_` version does skip the DB call, which is a nice side effect.